src/Data/Unfoldable.purs

```diff
@@ -88,6 +89,13 @@ none = unfoldr (const Nothing) unit
 singleton :: forall f a. Unfoldable f => a -> f a
 singleton = replicate 1
 
+-- | Produce a Lazy Unfoldable structure representing the natural numbers.
+nats :: forall f a. Unfoldable f => Semiring a => Lazy (f a) => f a
```

Comment:
`f` might be lazy but in the wrong way. I think we need a comment to say that the result is actually infinite.